openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -0,0 +1,179 @@
+            `requesting_user.username` is assumed.
+
+    Raises:
+        AccountUserNotFound: no user with username `username` exists (or `requesting_user.username` if
+            `username` is not specified)
+        AccountNotAuthorized: the requesting_user does not have access to change the account
+            associated with `username`
+        AccountUpdateError: the update could not be completed. Note that if multiple fields are updated at the same
+            time, some parts of the update may have been successful, even if an AccountUpdateError is returned;
+            in particular, the user account (not including e-mail address) may have successfully been updated,
+            but then the e-mail change request, which is processed last, may throw an error.
+    """
+    existing_user = _get_user(requesting_user, username)
+    for preference_key in update.keys():
+        preference_value = update[preference_key]
+        if preference_value:
+            set_user_preference(requesting_user, preference_key, preference_value, username=username)
+        else:
+            delete_user_preference(requesting_user, preference_key, username=username)
+
+
+def set_user_preference(requesting_user, preference_key, value, username=None):
+    """Update a user preference for the given username.
+
+    Note:
+        It is up to the caller of this method to enforce the contract that this method is only called
+        with the user who made the request.
+
+    Arguments:
+        requesting_user (User): The user requesting to modify account information. Only the user with username
+            'username' has permissions to modify account information.
+        preference_key (string): The key for the user preference.
+        value (string): The value to be stored.
+        username (string): Optional username specifying which account should be updated. If not specified,
+            `requesting_user.username` is assumed.
+
+    Raises:
+        AccountUserNotFound: no user with username `username` exists (or `requesting_user.username` if
+            `username` is not specified)
+        AccountNotAuthorized: the requesting_user does not have access to change the account
+            associated with `username`
+        AccountValidationError: the update was not attempted because validation errors were found with
+            the supplied update
+        AccountUpdateError: the update could not be completed. Note that if multiple fields are updated at the same
+            time, some parts of the update may have been successful, even if an AccountUpdateError is returned;
+            in particular, the user account (not including e-mail address) may have successfully been updated,
+            but then the e-mail change request, which is processed last, may throw an error.
+    """
+    existing_user = _get_user(requesting_user, username)
+    user_preference, __ = existing_user.preferences.get_or_create(user=existing_user, key=preference_key)
```

Comment:
I would use UserPreference.set_preference (unless our goal is to somehow get rid of that altogether).